tests: littlefs: multiple NXP platforms build fail due to missing partition compatibility in overlays

### Describe the bug

`tests/subsys/fs/littlefs` fails to build on multiple NXP platforms.

The failure is caused by board-specific littlefs test overlays that define
`large_partition`, `medium_partition`, and `small_partition`, but do not
mark those nodes with the partition compatibility expected by the current
board DTS model.

After commit `5d0c7b8df7318671be7e44ad2497b9fe91c47755`
("boards: nxp: use zephyr,mapped-partition for FlexSPI XIP backings"),
these board families use `zephyr,mapped-partition` for the relevant flash
partitions. The littlefs test overlays still add raw partition nodes, so
`PARTITION_ID(small_partition)` / `PARTITION_ID(large_partition)` in
`tests/subsys/fs/littlefs/src/testfs_lfs.c` fall through to the fixed
partition path and the build fails.

This issue was first reported from
`mimxrt1060_evk@C/mimxrt1062/qspi`, but the same failure signature is
present on several other NXP platforms.

### Regression

- [x] is_regression

Yes. Static analysis points to regression after:

5d0c7b8df7318671be7e44ad2497b9fe91c47755

https://github.com/zephyrproject-rtos/zephyr/pull/108207

### Steps to reproduce

1. Use Zephyr `main`
2. Build littlefs tests for an affected platform, for example:

```sh
scripts/twister -p mimxrt1060_evk@C/mimxrt1062/qspi \
  -T tests/subsys/fs/littlefs \
  --build-only --inline-logs
```

I originally saw this while running:

```sh
scripts/twister -j 12 -T tests -T samples \
  --build-only --inline-logs -M all \
  -p mimxrt1060_evk@C/mimxrt1062/qspi
```

### Relevant log output

```sh
include/zephyr/storage/flash_map.h:392:22: note: in expansion of macro 'DT_FIXED_PARTITION_ID'
include/zephyr/devicetree.h:5983:24: note: in expansion of macro 'DT_N_NODELABEL_small_partition'
include/zephyr/storage/flash_map.h:392:44: note: in expansion of macro 'DT_NODELABEL'
tests/subsys/fs/littlefs/src/testfs_lfs.c:13:33: note: in expansion of macro 'PARTITION_ID'
tests/subsys/fs/littlefs/src/testfs_lfs.c:25:32: note: in expansion of macro 'SMALL_PARTITION_ID'
ninja: build stopped: subcommand failed.
```

The same issue also shows up for `large_partition` in the custom littlefs
variants.

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- OS: Windows host, Linux-based CI reproduction
- Toolchain: Zephyr toolchain in CI
- Commit SHA used when reported: `ab92232cfcd1362ee481ba06197fda7d6613aff5`
- Suspected regression introduced by:
  `5d0c7b8df7318671be7e44ad2497b9fe91c47755`

### Additional Context

Affected platforms observed from CI failure artifacts:

- `frdm_mcxn947/mcxn947/cpu0`
- `mimxrt1010_evk/mimxrt1011`
- `mimxrt1015_evk/mimxrt1015`
- `mimxrt1020_evk/mimxrt1021`
- `mimxrt1060_evk@C/mimxrt1062/qspi`
- `mimxrt1160_evk/mimxrt1166/cm7`
- `mimxrt1170_evk@B/mimxrt1176/cm7`
- `mimxrt1180_evk/mimxrt1189/cm33`
- `mimxrt595_evk/mimxrt595s/cm33`

Relevant files include:

- `tests/subsys/fs/littlefs/src/testfs_lfs.c`
- `tests/subsys/fs/littlefs/boards/mimxrt1010_evk.overlay`
- `tests/subsys/fs/littlefs/boards/mimxrt1015_evk.overlay`
- `tests/subsys/fs/littlefs/boards/mimxrt1020_evk.overlay`
- `tests/subsys/fs/littlefs/boards/mimxrt1060_evk_mimxrt1062_qspi_C.overlay`
- `tests/subsys/fs/littlefs/boards/mimxrt1160_evk_mimxrt1166_cm7.overlay`
- `tests/subsys/fs/littlefs/boards/mimxrt1170_evk_mimxrt1176_cm7.overlay`
- `tests/subsys/fs/littlefs/boards/mimxrt1180_evk_mimxrt1189_cm33.overlay`
- `tests/subsys/fs/littlefs/boards/mimxrt595_evk_mimxrt595s_cm33.overlay`
- `tests/subsys/fs/littlefs/boards/frdm_mcxn947_mcxn947_cpu0.overlay`

Observations from source review:

- Several NXP littlefs test overlays recreate partition nodes without
  `compatible = "zephyr,mapped-partition";`.
- Base board DTS files for the affected boards already use mapped
  partitions.
- `PARTITION_ID(label)` in `include/zephyr/storage/flash_map.h` therefore
  resolves down the wrong path for these overlays.

A likely fix is to make the affected littlefs test overlays define
compatible partition nodes matching the current board DTS scheme.

--- a/boards/nxp/mimxrt1170_evk/mimxrt1170_evk.dtsi
+++ b/boards/nxp/mimxrt1170_evk/mimxrt1170_evk.dtsi
@@ -297,6 +297,10 @@ arduino_spi: &lpspi1 {
 			erase-block-size = <DT_SIZE_K(4)>;
 			write-block-size = <1>;
 
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges;
+
 			partitions {
 				#address-cells = <1>;
 				#size-cells = <1>;

--- a/boards/nxp/mimxrt1170_evk/mimxrt1170_evk_mimxrt1176_cm7_B.overlay
+++ b/boards/nxp/mimxrt1170_evk/mimxrt1170_evk_mimxrt1176_cm7_B.overlay
@@ -40,6 +40,10 @@
 			erase-block-size = <DT_SIZE_K(4)>;
 			write-block-size = <1>;
 
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges;
+
 			partitions {
 				#address-cells = <1>;
 				#size-cells = <1>;

--- a/boards/nxp/rd_rw612_bga/rd_rw612_bga.dtsi
+++ b/boards/nxp/rd_rw612_bga/rd_rw612_bga.dtsi
@@ -172,6 +172,10 @@ arduino_i2c: &flexcomm2 {
 			erase-block-size = <DT_SIZE_K(4)>;
 			write-block-size = <1>;
 
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges;
+
 			partitions {
 				#address-cells = <1>;
 				#size-cells = <1>;

--- a/tests/subsys/fs/littlefs/boards/frdm_mcxn947_mcxn947_cpu0.overlay
+++ b/tests/subsys/fs/littlefs/boards/frdm_mcxn947_mcxn947_cpu0.overlay
@@ -9,16 +9,19 @@
 &w25q64jvssiq {
 	partitions {
 		large_partition: partition@400000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x00400000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@700000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x00700000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@7f0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x007f0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/frdm_rw612.overlay
+++ b/tests/subsys/fs/littlefs/boards/frdm_rw612.overlay
@@ -9,16 +9,19 @@
 &w25q512jvfiq {
 	partitions {
 		large_partition: partition@3c00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x03c00000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@3f00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x03f00000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@3ff0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x03ff0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt1010_evk.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt1010_evk.overlay
@@ -9,16 +9,19 @@
 &at25sf128a {
 	partitions {
 		large_partition: partition@c00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x00c00000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@f00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x00f00000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@ff0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x00ff0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt1015_evk.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt1015_evk.overlay
@@ -9,16 +9,19 @@
 &at25sf128a {
 	partitions {
 		large_partition: partition@c00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x00c00000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@f00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x00f00000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@ff0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x00ff0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt1020_evk.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt1020_evk.overlay
@@ -9,16 +9,19 @@
 &is25lp064 {
 	partitions {
 		large_partition: partition@400000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x00400000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@700000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x00700000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@7f0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x007f0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt1024_evk.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt1024_evk.overlay
@@ -9,6 +9,7 @@
 &w25q32jvwj0 {
 	partitions {
 		small_partition: partition@3f0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x003f0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt1040_evk.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt1040_evk.overlay
@@ -9,16 +9,19 @@
 &w25q64jvssiq {
 	partitions {
 		large_partition: partition@400000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x00400000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@700000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x00700000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@7f0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x007f0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt1060_evk_mimxrt1062_qspi_C.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt1060_evk_mimxrt1062_qspi_C.overlay
@@ -9,16 +9,19 @@
 &w25q128jw {
 	partitions {
 		large_partition: partition@c00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x00c00000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@f00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x00f00000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@ff0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x00ff0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt1064_evk.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt1064_evk.overlay
@@ -9,16 +9,19 @@
 &is25wp064 {
 	partitions {
 		large_partition: partition@400000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x00400000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@700000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x00700000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@7f0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x007f0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt1160_evk_mimxrt1166_cm7.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt1160_evk_mimxrt1166_cm7.overlay
@@ -9,16 +9,19 @@
 &is25wp128 {
 	partitions {
 		large_partition: partition@c00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x00c00000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@f00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x00f00000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@ff0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x00ff0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt1170_evk_mimxrt1176_cm7.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt1170_evk_mimxrt1176_cm7.overlay
@@ -9,16 +9,19 @@
 &w25q512nw {
 	partitions {
 		large_partition: partition@3c00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x03c00000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@3f00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x03f00000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@3ff0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x03ff0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt1180_evk_mimxrt1189_cm33.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt1180_evk_mimxrt1189_cm33.overlay
@@ -9,16 +9,19 @@
 &w25q128jw {
 	partitions {
 		large_partition: partition@c00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x00c00000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@f00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x00f00000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@ff0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x00ff0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt595_evk_mimxrt595s_cm33.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt595_evk_mimxrt595s_cm33.overlay
@@ -9,16 +9,19 @@
 &mx25um51345g {
 	partitions {
 		large_partition: partition@3c00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x03c00000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@3f00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x03f00000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@3ff0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x03ff0000 DT_SIZE_K(64)>;
 		};

--- a/tests/subsys/fs/littlefs/boards/mimxrt685_evk_mimxrt685s_cm33.overlay
+++ b/tests/subsys/fs/littlefs/boards/mimxrt685_evk_mimxrt685s_cm33.overlay
@@ -9,16 +9,19 @@
 &mx25um51345g {
 	partitions {
 		large_partition: partition@3c00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "large";
 			reg = <0x03c00000 DT_SIZE_M(3)>;
 		};
 
 		medium_partition: partition@3f00000 {
+			compatible = "zephyr,mapped-partition";
 			label = "medium";
 			reg = <0x03f00000 DT_SIZE_K(960)>;
 		};
 
 		small_partition: partition@3ff0000 {
+			compatible = "zephyr,mapped-partition";
 			label = "small";
 			reg = <0x03ff0000 DT_SIZE_K(64)>;
 		};

